src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_AMOUNT = FigureName.values().length;
+    private static final int MAX_SIZE = 10;
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    Random random = new Random();
+    ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureName figureName = FigureName.values()[random.nextInt(FIGURES_AMOUNT)];
+        switch (figureName) {
+            case CIRCLE -> {
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(MAX_SIZE));
+            }
+            case RECTANGLE -> {
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIZE), random.nextInt(MAX_SIZE));
+            }
+            case RIGHT_TRIANGLE -> {
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIZE), random.nextInt(MAX_SIZE));
+            }
+            case SQUARE -> {
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(MAX_SIZE));
+            }
+            case ISOSCELES_TRAPEZOID -> {
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIZE), random.nextInt(MAX_SIZE),
+                        random.nextInt(MAX_SIZE));
+            }
+            default -> {
+                return new Circle(DEFAULT_COLOR.toString(), MAX_SIZE);
```

Comment:
Don't return 'null' from a method. Instead of using a default case that creates a specific figure, you could consider throwing an exception to indicate that an unsupported figure type was encountered, or handle all possible cases explicitly without a default case.